src/DynamicRupture/Output/ReceiverBasedOutput.cpp

```diff
@@ -0,0 +1,426 @@
+
+  local.transientNormalTraction =
+      local.faceAlignedValuesPlus[0] +
+      ((diff(0) + impAndEta.zpNeig * diff(6)) * impAndEta.zp) * norDivisor;
+
+  local.faultNormalVelocity =
+      local.faceAlignedValuesPlus[6] +
+      (local.transientNormalTraction - local.faceAlignedValuesPlus[0]) * impAndEta.invZp;
+
+  real missingSigmaValues = (local.transientNormalTraction - local.faceAlignedValuesPlus[0]);
+  missingSigmaValues *= (1.0 - 2.0 * std::pow(local.waveSpeedsPlus->sWaveVelocity /
+                                                  local.waveSpeedsPlus->pWaveVelocity,
+                                              2));
+
+  local.faceAlignedStress22 = local.faceAlignedValuesPlus[1] + missingSigmaValues;
+  local.faceAlignedStress33 = local.faceAlignedValuesPlus[2] + missingSigmaValues;
+  local.faceAlignedStress23 = local.faceAlignedValuesPlus[4];
+}
+
+void ReceiverBasedOutput::updateLocalTractions(real strength) {
+  auto squaredComponent1 = std::pow(local.iniTraction1 + local.faceAlignedStress12, 2);
+  auto squaredComponent2 = std::pow(local.iniTraction2 + local.faceAlignedStress13, 2);
+  auto tracEla = std::sqrt(squaredComponent1 + squaredComponent2);
+
+  if (tracEla > std::abs(strength)) {
+    local.updatedTraction1 =
+        ((local.iniTraction1 + local.faceAlignedStress12) / tracEla) * strength;
+    local.updatedTraction2 =
+        ((local.iniTraction2 + local.faceAlignedStress13) / tracEla) * strength;
+
+    // update stress change
+    local.updatedTraction1 -= local.iniTraction1;
+    local.updatedTraction2 -= local.iniTraction2;
+  } else {
+    local.updatedTraction1 = local.faceAlignedStress12;
+    local.updatedTraction2 = local.faceAlignedStress13;
+  }
+}
+
+void ReceiverBasedOutput::computeSlipRate(std::array<real, 6>& rotatedTraction,
+                                          std::array<real, 6>& rotatedLocalStress) {
+
+  auto& impAndEta = ((local.layer->var(drDescr->impAndEta))[local.ltsId]);
+  local.slipRateStrike = -impAndEta.invEtaS * (rotatedTraction[3] - rotatedLocalStress[3]);
+  local.slipRateDip = -impAndEta.invEtaS * (rotatedTraction[5] - rotatedLocalStress[5]);
+}
+
+void ReceiverBasedOutput::computeSlipRate(const double* tangent1,
+                                          const double* tangent2,
+                                          const double* strike,
+                                          const double* dip) {
+  local.slipRateStrike = static_cast<real>(0.0);
+  local.slipRateDip = static_cast<real>(0.0);
+
+  for (size_t i = 0; i < 3; ++i) {
+    real factorMinus = (local.faceAlignedValuesMinus[7] * tangent1[i] +
+                        local.faceAlignedValuesMinus[8] * tangent2[i]);
+
+    real factorPlus = (local.faceAlignedValuesPlus[7] * tangent1[i] +
+                       local.faceAlignedValuesPlus[8] * tangent2[i]);
+
+    local.slipRateStrike += factorMinus * strike[i] - factorPlus * strike[i];
+    local.slipRateDip += factorMinus * dip[i] - factorPlus * dip[i];
```

Comment:
(factorMinus-factorPlus)*dip[i];

Reply:
done (line 328 and a similar thing for line 327)